vignettes/datatable-importing.Rmd

```diff
@@ -195,6 +195,115 @@ For more canonical documentation of defining packages dependency check the offic
+
+#### Importing Specific Functions and Symbols
+
+If you use specific functions like `data.table()`, `setDT()`, `merge()`, or special symbols like `.N`, `.SD`, `:=`, import them individually:
+```r
+importFrom(data.table, data.table, setDT, merge, .N, .SD, :=)
+```
+
+#### Importing S3 Generics
+
+If your package adds methods to any S3 generics owned by `data.table`, you need to import these generics. Check the `data.table` NAMESPACE file for the latest set of S3 generics:
+```r
+importFrom(data.table, as.data.table, groupingsets, as.IDate)
+```
+
+#### Importing S4 Classes
+
+If your package uses S4 classes from `data.table`, you need to import these classes. This is less common but necessary for advanced use cases:
+```r
+importClassesFrom(data.table, data.table, IDate, ITime)
+```
+
+#### Blanket Import
+
+Alternatively, you can import all functions from `data.table` at once, though this is generally not recommended:
+
+```r
+import(data.table)
+```
+
+**Justification for Avoiding Blanket Imports:**
+1. **Documentation**: The NAMESPACE file can serve as good documentation of how you depend on certain packages.
+2. **Avoiding Conflicts**: Blanket imports leave you open to subtle breakage. For example, if you import(pkgA) and import(pkgB), but later pkgB exports a function also exported by pkgA, this will break your package due to conflicts in your namespace, which is disallowed by `R CMD check` and CRAN.
+
+### Step 3: Update Your R code Files outside package's R/ directory
+
+When you move a package from `Depends` to `Imports`, it will no longer be automatically attached when your package is loaded. However, if you correctly import `data.table` in your NAMESPACE, you should not need to make changes to your package's R code files. This is different from user scripts or vignettes, which may need adjustments.
+
+#### For User Scripts and Vignettes
+
+If you are working with user scripts, vignettes, or other R files outside the package's R/ directory (e.g., tests and demos), you might need to explicitly load `data.table`.
+
+**Before (with `Depends`):**
+```r
+# data.table functions are directly available
+library(MyPkgDependsDataTable)
+dt <- data.table(x = 1:10, y = letters[1:10])
+setDT(dt)
+result <- merge(dt, other_dt, by = "x")
+```
+
+**After (with `Imports`):**
+```r
+# Explicitly load data.table in user scripts or vignettes
+library(data.table)
+library(MyPkgDependsDataTable)
+dt <- data.table(x = 1:10, y = letters[1:10])
+setDT(dt)
+result <- merge(dt, other_dt, by = "x")
+```
+
+### Additional Considerations
+
+- **Special Symbols**: Ensure `data.table` symbols like `.N`, `.SD`, and `:=` are imported.
+- **Function Usage**: Use tools like `codetools` or `devtools` to analyze function usage and ensure all necessary imports are covered.
+- **S3 and S4 Methods**: Ensure all necessary generics and classes are imported if your package extends `data.table` functionality with custom methods.
```

Comment:
I'm not sure this section adds anything to the above, WDYT about removing it?

Reply:
sure 👍